apps/server/src/services/auth0Service.ts

```diff
@@ -0,0 +1,41 @@
+import config from '../config'
+import { warning as logWarning } from '../infra/logger'
+import { get, set } from './storageClient/redis'
+
+const REDIS_AUTH_ACCESS_KEY = 'auth0AccessToken'
+
+export async function getAccessToken() {
+  const { UCPClientId, UCPClientSecret } = config
+
+  const cachedAccessToken = await get(REDIS_AUTH_ACCESS_KEY)
+  if (cachedAccessToken) {
+    return cachedAccessToken
+  }
+
+  try {
+    const response = await fetch(config.Auth0TokenUrl, {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json'
+      },
+      body: JSON.stringify({
+        grant_type: 'client_credentials',
+        audience: 'ucp-widget-interactions',
+        client_id: UCPClientId,
+        client_secret: UCPClientSecret
+      })
+    })
+    if (response.status === 401) {
+      logWarning('Unauthorized to retrieve UCP access token')
+      return null
+    }
+    const responseJson = await response.json()
+    await set(REDIS_AUTH_ACCESS_KEY, responseJson.access_token, {
+      EX: responseJson.expires_in
+    })
+
+    return responseJson.access_token
+  } catch (error) {
+    throw new Error('Could not get UCP access token', error)
```

Comment:
Is this triggering because response.json fails? I don't think fetch would be throwing an actual error